web/base_pdf_page_view.js

```diff
@@ -100,10 +112,41 @@ class BasePDFPageView {
     // have a final flash we just display it once all the drawing is done.
     const updateOnFirstShow = !prevCanvas && !hasHCM && !hideUntilComplete;
 
-    const canvas = (this.canvas = document.createElement("canvas"));
+    let canvas = (this.canvas = document.createElement("canvas"));
 
     this.#showCanvas = isLastShow => {
+      let tempCanvas = this.#tempCanvas;
       if (updateOnFirstShow) {
+        if (!isLastShow && this.#minDurationToUpdateCanvas > 0) {
```

Comment:
The new variable probably doesn't need to be unconditionally defined, since it's only used when `updateOnFirstShow === true`?
```suggestion
      if (updateOnFirstShow) {
        let tempCanvas = this.#tempCanvas;
        if (!isLastShow && this.#minDurationToUpdateCanvas > 0) {
```